cdap-ui/app/cdap/components/DataPrep/Directives/Format/index.js

```diff
@@ -123,54 +86,340 @@ export default class Format extends Component {
+    if (!this.state.createNewColumn) { return null; }
+
+    return (
+      <div>
+        <Input
+          type="text"
+          className="form-control mousetrap"
+          value={this.state.newColumnInput}
+          onChange={this.setNewColumnInput}
+          placeholder={T.translate(`${COPY_NEW_COLUMN_PREFIX}.inputPlaceholder`)}
+          autoFocus
+        />
+        {
+          columnNameAlreadyExists(this.state.newColumnInput) ? (
+            <WarningContainer
+              message={T.translate(`${COPY_NEW_COLUMN_PREFIX}.inputDuplicate`)}
+            />
+          ) : null
+        }
+      </div>
+    );
+  }
+
+  renderFormatPopover() {
+    if (!this.state.formatPopoverOpen) { return null; }
 
     return (
       <div
-        className="parse-detail second-level-popover"
-        onClick={this.preventPropagation}
+        className="third-level-popover"
+        onClick={preventPropagation}
       >
-        <div className="parse-options">
+        <div>
+          {T.translate(`${PREFIX}.Formats.CONCATENATE.addLabel`)}
+        </div>
+        <Input
+          className="format-input mousetrap"
+          value={this.state.formatInput}
+          placeholder={T.translate(`${PREFIX}.Formats.CONCATENATE.inputPlaceholder`)}
+          onChange={this.setFormatInput}
+          autoFocus
+        />
+        <Input
+          type="select"
+          className="concatenate-option-select"
+          onChange={this.setConcatOption}
+          value={this.state.concatOption}
+        >
           {
-            this.FORMAT_OPTIONS.map((option) => {
+            this.CONCATENATE_OPTIONS.map(option => {
               return (
-                <div
-                  key={option.name}
-                  className="option"
-                  onClick={option.onClick}
-                >
-                  {T.translate(`${PREFIX}.Formats.${option.name}.label`)}
-                </div>
+                <option value={option.name}>
+                  {option.label}
+                </option>
               );
             })
           }
+        </Input>
+        <div>
+          {T.translate(`${PREFIX}.Formats.CONCATENATE.addDescription`)}
         </div>
+
+        <div
+          className="create-new-column-line"
+          onClick={this.toggleCreateNewColumn}
+        >
+          <span className="fa fa-fw">
+            <IconSVG
+              name={this.state.createNewColumn ? 'icon-check-square' : 'icon-square-o'}
+            />
+          </span>
+
+          <span>
+            {T.translate(`${COPY_NEW_COLUMN_PREFIX}.label`)}
+          </span>
+        </div>
+
+        {this.renderNewColumnNameInput()}
+
+        <hr />
+
+        {this.renderActionButtons()}
       </div>
     );
   }
-  render() {
+
+  renderModal() {
+    return this.state.activeModal;
+  }
+  renderOptions() {
+    return (
+      this.FORMAT_OPTIONS
+      .filter(option => option.validColTypes.indexOf(this.columnType) !== -1)
```

Comment:
nit: indentation? I feel we should start using prettier to avoid formatting issues anymore. Will see if I can add a post-commit hook (might get knocked out by backend guys going through the same thing 😄 )